edstem-api/edstem_api/macros.py

```diff
@@ -0,0 +1,233 @@
+        },
+        username=username,
+        password=password,
+    )
+
+    return result is not None
+
+
+def users(course_id, username=None, password=None):
+    result = api.request(
+        endpoint="courses/{}/admin".format(course_id),
+        username=username,
+        password=password,
+    )
+
+    if "users" in result:
+        return result.get("users", [])
+
+def get_feed(course_id, limit=-1, sort="date", order="desc", username=None, password=None):
+    """
+    """
+    if limit > -1:
+        result = api.request(
+            endpoint=f"courses/{course_id}/threads?sort={sort}&order={order}",
+            username=username,
+            password=password,
+        )
+    else:
+        result = api.request(
+            endpoint=f"courses/{course_id}/threads?limit={limit}&sort={sort}&order={order}",
+            username=username,
+            password=password,
+        )
+    return result
+
+
+def get_thread_content(thread_id, username=None, password=None):
+    result = api.request(
+        endpoint=f"threads/{thread_id}",
+        username=username,
+        password=password,
+        )
+    return result
+
+
+def simplify_thread_info(thread: dict) -> dict:
+    """Disseminates a thread into useful, easy-to-understand information.
+
+    Args:
+        thread (dict): JSON-styled dict representing a thread.
+    
+    Returns:
+        dict: Simplified thread with the useful info.
+    """
+
+    # We define a simple helper function here, to recurse through the entire 
+    # post "tree" structure and apply any post-changing functions to the post.
+    # Very helpful for dealing with nested answers and comments of a thread.
+    def recurse_on_post(f, post: dict, **kwargs):
+        post = f(post, **kwargs)
+
+        if "answers" in post:
+            new_answers = []
+            for answer in post["answers"]:
+                new_answer = recurse_on_post(f, answer, **kwargs)
+                new_answers.append(new_answer)
+            post["answers"] = new_answers
+
+        if "comments" in post:
+            new_comments = []
+            for comment in post["comments"]:
+                new_comment = recurse_on_post(f, comment, **kwargs)
+                new_comments.append(new_comment)
+            post["comments"] = new_comments
+
+        return post
+
+    # Slice a subset of the original post contents with the keys we want.
+    def simplify_post(post: dict, keys: set) -> dict:
+        return {x: post[x] for x in post if x in keys}
+
+    # Exchange meaningless numbers with actual names of people.
+    def map_ids_to_names(post: dict, users: list) -> dict:
+        user_id = post["user_id"]
+        for user in users:
+            if user["id"] == user_id:
+                post["user_id"] = user["name"]
+                # Rename the key (and preserve dict order) for clarity
+                post = {"by" if k == "user_id" else k:v for k,v in post.items()}
+                break
+
+        return post
```

Comment:
Minor thing, but we probably don't actually care who said what. We only care about their role. It might be worth changing this in another iteration